llvm/lib/Target/AMDGPU/SIFoldOperands.cpp

```diff
@@ -1092,7 +1092,8 @@ void SIFoldOperandsImpl::foldOperand(
   } else {
     if (UseMI->isCopy() && OpToFold.isReg() &&
         UseMI->getOperand(0).getReg().isVirtual() &&
-        !UseMI->getOperand(1).getSubReg()) {
+        !UseMI->getOperand(1).getSubReg() &&
+        !OpToFold.getParent()->hasRegisterImplicitUseOperand(AMDGPU::EXEC)) {
```

Comment:
Our use of implicit exec operands on copies is super unsound and we should stop doing it. That being said, I think this should just generalize to disallowing handling a copy with any implicit operands